File: TemplateLayout/TimerViewController.m
//
//  TimerViewController.m
//  TemplateLayout
//
//  Created by 哈哈 on 2020/4/8.
//  Copyright © 2020 MengHeng. All rights reserved.
//

#import "TimerViewController.h"

@interface TimerViewController ()
@property (nonatomic,strong) NSTimer *timer;
@end

@implementation TimerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    NSTimer *timer =[ NSTimer timerWithTimeInterval:1.0f repeats:true block:^(NSTimer * _Nonnull timer) {
        NSLog(@"调用了");
    }];
    NSTimer *timer1 =[NSTimer timerWithTimeInterval:0.5 target:self selector:@selector(action:) userInfo:nil repeats:YES];
//    self.timer = timer1;
//
//    [timer1 fire];
    [[NSRunLoop currentRunLoop] addTimer:timer1 forMode:NSRunLoopCommonModes];
}

-(void)action:(NSTimer *)timer{
    NSLog(@"定时器开");
}

-(void)viewWillDisappear:(BOOL)animated{
    [self.timer invalidate];
    self.timer = nil;
}


@end

File: mOVe/mOVe/Timer/TimerViewController.swift
//
//  TimerViewController.swift
//  mOVe
//
//  Created by 哈哈 on 2022/6/7.
//  Copyright © 2022 MengHeng. All rights reserved.
//

import UIKit

/// 定时器的相关操作
class TimerViewController: UIViewController {
    var killTimer:MHSecondKillManager!

    override func viewDidLoad() {
        super.viewDidLoad()
        killTimer = MHSecondKillManager(seconds: 12 * 60 * 60, callBack: {[weak self] (text) in
                 
        })
               
        
    }
}
